agent/reconciler/host_reconciler.go

```diff
@@ -231,7 +231,11 @@ func (r *HostReconciler) cleank8sdirectories(ctx context.Context) error {
 	}
 
 	if len(errList) > 0 {
-		return errors.New("not all k8s directories are cleaned up")
+		errs := errList[0]                //nolint: gosec
+		for i, err := range errList[1:] { //nolint: gosec
+			errs = fmt.Errorf("%w; %d error", err, i)
```

Comment:
This would overwrite the previous errors in `errs`, keeping the last one only.

Reply:
Good catch, Pushed a new commit.